packages/cal-to-sheet/cal-to-sheet.js

```diff
@@ -0,0 +1,89 @@
+const die = msg => {
+  throw Error(msg);
+};
+const NonNullish = (x, msg) => x || die(msg);
+
+function getCalendarEventUrl(calendarId, eventId) {
+  return `https://calendar.google.com/calendar/event?cid=${calendarId}&eid=${eventId}`;
+}
+
+/**
+ * Updates the spreadsheet with events from the specified Google Calendar.
+ */
+function updateSheetFromCalendar(_nonce, io = {}) {
+  const {
+    // Get the spreadsheet and sheet
+    ss = SpreadsheetApp.getActiveSpreadsheet(),
+    sheet = ss.getActiveSheet() || ss.getSheets()[0],
+    sheetName = sheet.getName(),
+    // Get the calendar
+    calendar = NonNullish(
+      CalendarApp.getCalendarsByName(sheetName),
+      `cannot find calendar ${sheetName}`,
+    )[0],
+    // Get today's date (for the start of the query)
+    today = new Date(),
+    ui = SpreadsheetApp.getUi(),
+  } = io;
+
+  const calendarId = calendar.getId();
+
+  const firstRow = 1; // The first row to write data to (header row assumed above)
+  const clearExistingData = true; // Set to true to clear existing data before updating
+
+  today.setHours(0, 0, 0, 0); // Start of the day
+
+  // Get events from the calendar (you can adjust the date range as needed)
+  const events = calendar.getEvents(
+    today,
+    new Date(today.getTime() + 365 * 24 * 60 * 60 * 1000),
+  ); // Events for the next year
```

Comment:
I'd rather get all the events, past or future, but only 1 for repeating events. Like a .ics export.